toolbox/include/base/RobotInterface.h

```diff
@@ -32,225 +35,118 @@ namespace iDynTree {
 
 namespace wbt {
     class RobotInterface;
-    struct YarpInterfaces;
-
-    typedef int jointIdx_yarp;
-    typedef int jointIdx_iDynTree;
-    typedef unsigned cb_idx;
-    typedef unsigned max_cb_idx;
-    typedef unsigned controlledJointIdxCB;
-    typedef std::unordered_map<jointIdx_iDynTree, std::pair<cb_idx, jointIdx_yarp>> JointsMapIndex;
-    typedef std::unordered_map<std::string, std::pair<cb_idx, jointIdx_yarp>> JointsMapString;
-    typedef std::unordered_map<std::string, controlledJointIdxCB> ControlledJointsMapCB;
-    typedef std::unordered_map<cb_idx, max_cb_idx> ControlBoardIdxLimit;
+    class Configuration;
+
+    using JointIndex_Yarp = int;
+    using JointIndex_iDynTree = int;
+    using JointName = std::string;
+    using ControlBoardIndex = unsigned;
+    using JointIndexInControlBoard = unsigned;
+
+    using JointIndexToYarpMap =
```

Comment:
I am a bit confused by all this structures (and avoiding this is actually why the remapper and the support for specifying the serialization in iDynTree was explicitly coded to avoid this constructs!). 
If this is just used for the "smandruppator", in the long run I suggests to rename this variables to be a generic "subpart" division of the joints, as in the future we could easily connect to the robot using a single controlboard containing all the joints (for performance reason) while keeping the nice division of the robot in parts for visualization/debugging.

Reply:
All these maps and method for gathering them are for the "smandruppator" (ModelPartitioner). I plan sooner or later to polish this RobotInterface class further and use it as standalone wrapper for the RemoteControlBoardRemapper and KinDynComputations. Since these maps might be helpful and coding them was a real pain, I decided to store them here. They are lazily accessed, if you never ask the get methods the code that spawns multiple single control board is never executed.